CreateService/handler.ts

```diff
@@ -159,45 +160,52 @@ export function CreateServiceHandler(
     context.log.info(
       `${logPrefix}| Creating new service with subscriptionId=${subscriptionId}`
     );
-    return createSubscriptionTask(
-      getLogger(context, logPrefix, "CreateSubscription"),
-      apiClient,
-      userAttributes.email,
-      subscriptionId,
-      productName
-    )
-      .chain(subscription =>
-        getUserTask(
-          getLogger(context, logPrefix, "GetUser"),
-          apiClient,
-          userAttributes.email
-        ).chain(userInfo =>
-          createServiceTask(
-            getLogger(context, logPrefix, "CreateService"),
+    return pipe(
+      createSubscriptionTask(
+        getLogger(context, logPrefix, "CreateSubscription"),
+        apiClient,
+        userAttributes.email,
+        subscriptionId,
+        productName
+      ),
+      TE.chain(subscription =>
+        pipe(
+          getUserTask(
+            getLogger(context, logPrefix, "GetUser"),
             apiClient,
-            servicePayload,
-            subscriptionId,
-            (sandboxFiscalCode as unknown) as FiscalCode,
-            userInfo.token_name
-          ).map(service => {
-            telemetryClient.trackEvent({
-              name: "api.services.create",
-              properties: {
-                isVisible: String(service.is_visible),
-                requesterUserEmail: userAttributes.email,
-                subscriptionId
-              }
-            });
-            return ResponseSuccessJson({
-              ...service,
-              primary_key: subscription.primary_key,
-              secondary_key: subscription.secondary_key
-            });
-          })
+            userAttributes.email
+          ),
+          TE.chain(userInfo =>
+            pipe(
```

Comment:
Instead of 3 level `chain` nested `pipe` functions, we can have a flatter and maybe more readable version using the following:
```typescript
return pipe(
      sequenceS(TE.ApplicativePar)({
        subscription: createSubscriptionTask(
          getLogger(context, logPrefix, "CreateSubscription"),
          apiClient,
          userAttributes.email,
          subscriptionId,
          productName
        ),
        user: getUserTask(
          getLogger(context, logPrefix, "GetUser"),
          apiClient,
          userAttributes.email
        )
      }),
      TE.bind("service", ({ user }) =>
        createServiceTask(
          getLogger(context, logPrefix, "CreateService"),
          apiClient,
          servicePayload,
          subscriptionId,
          (sandboxFiscalCode as unknown) as FiscalCode,
          user.token_name
        )
      ),
      TE.map(({ service, subscription }) => {
        telemetryClient.trackEvent({
          name: "api.services.create",
          properties: {
            isVisible: String(service.is_visible),
            requesterUserEmail: userAttributes.email,
            subscriptionId
          }
        });
        return ResponseSuccessJson({
          ...service,
          primary_key: subscription.primary_key,
          secondary_key: subscription.secondary_key
        });
      }),
      TE.toUnion
    )();
```
If we want to keep sequentiality we can use `TE.ApplicativeSeq`
cc: @AleDore @balanza

Reply:
love it, thanks!

Reply:
it only works with `TE.ApplicativeSeq` actually

Reply:
Yes, it because of the missing getUser mock into the tests.

Reply:
Right, I see. Mocks are defined test by test, so calling it in parallel causes them to be called and fail. I'll fix it, thanks

Reply:
Done [here](https://github.com/pagopa/io-functions-services/pull/154/commits/e9611ff18d93b5d9ff05c7d53c0f701829eb0196)